File: 프로그래머스/unrated/181857. 배열의 길이를 2의 거듭제곱으로 만들기/배열의 길이를 2의 거듭제곱으로 만들기.cpp
#include <string>
#include <vector>
#include <math.h>
using namespace std;

vector<int> solution(vector<int> arr) {
    int cnt = 1;
    
    while (cnt < arr.size())
        cnt *= 2;
    
    arr.resize(cnt, 0);
    
    return arr;
}

File: 프로그래머스/0/181857. 배열의 길이를 2의 거듭제곱으로 만들기/배열의 길이를 2의 거듭제곱으로 만들기.java
import java.util.Arrays;

class Solution {
    public int[] solution(int[] arr) {
        if (arr.length <= 2) return Arrays.copyOf(arr, arr.length);

        double log = Math.log(arr.length) / Math.log(2);
        int ceil = (int) Math.ceil(log);
        int exp = (int) Math.pow(2, ceil);
        return Arrays.copyOf(arr, exp);
    }
}